src/puppetlabs/trapperkeeper/internal.clj

```diff
@@ -448,20 +475,24 @@
         this)
       (a/stop [this]
         (shutdown! app-context)
-        this))))
+        this)
+      (a/restart [this]
+        (run-lifecycle-fns app-context s/stop "stop" (reverse ordered-services))
```

Comment:
It makes sense to me to shut the app process down in that case, but maybe @cprice404 has thoughts here.